include/rocksdb/advanced_options.h

```diff
@@ -91,11 +91,37 @@ struct CompressionOptions {
   //
   // The default value is the max 16-bit int as it'll be written out in OPTIONS
   // file, which should be portable.
-  const static int kDefaultCompressionLevel = 32767;
-
-  int window_bits;
-  int level;
-  int strategy;
+  static constexpr int kDefaultCompressionLevel = 32767;
+
+  // zlib only: windowBits parameter. See https://www.zlib.net/manual.html
+  int window_bits = -14;
+
+  // Compression "level" applicable to zstd, zlib, LZ4. Except for
+  // kDefaultCompressionLevel (see above), the meaning of each value depends
+  // on the compression algorithm.
+  int level = kDefaultCompressionLevel;
+
+  // zlib only: strategy parameter. See https://www.zlib.net/manual.html
+  int strategy = 0;
+
+  // Essentially specifies a minimum acceptable compression ratio. A block is
+  // stored uncompressed if the compressed block does not achieve this ratio,
+  // because the downstream cost of decompression is not considered worth such
+  // a small savings (if any).
+  // However, the ratio is specified in a way that is efficient for checking.
+  // An integer from 1 to 1024 indicates the maximum allowable compressed bytes
+  // per 1KB of input, so the minimum acceptable ratio is 1024.0 / this value.
+  // For example, for a minimum ratio of 1.5:1, set to 683. See SetMinRatio().
+  // Default: abandon use of compression for a specific block or entry if
+  // compressed by less than 12.5% (minimum ratio of 1.143:1).
+  int max_compressed_bytes_per_kb = 1024 * 7 / 8;
```

Comment:
My interpretation was that the new syntax for configuring CompressionOptions is much preferred for clarity, consistency, and precision (override only the fields you want to override), and that people only extended the old syntax because that was the easy path to adding a new option. I'm trying to set a precedent that we don't need to continue extending that, and slowly compel users to upgrade their syntax when they want to use a new option.

But I guess your point is that it can confuse the meaning of the old syntax (support likely to continue for a long time) if we insert fields in between those. I can put the field after and make clear in the API docs the group of fields for which the old syntax is supported. 👍